Crash: address of invalid type

```c
void foo(void) {
    char x[64];
    char *y = &(x&x);
}
```

```
panic: reached unreachable code
/Users/ehaas/source/arocc/src/aro/Type.zig:2338:21: 0x10221b76c in fromType (arocc)
            else => unreachable,
                    ^
/Users/ehaas/source/arocc/src/aro/Type.zig:2482:48: 0x1022085f1 in printPrologue__anon_15488 (arocc)
        else => try w.writeAll(Builder.fromType(ty).str(langopts).?),
                                               ^
/Users/ehaas/source/arocc/src/aro/Type.zig:2429:49: 0x1022061c7 in printPrologue__anon_15488 (arocc)
        const simple = try elem_ty.printPrologue(mapper, langopts, w);
                                                ^
/Users/ehaas/source/arocc/src/aro/Type.zig:2404:29: 0x10215197c in print__anon_9247 (arocc)
    _ = try ty.printPrologue(mapper, langopts, w);
                            ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:464:16: 0x102217d49 in typePairStrExtra (arocc)
    try b.print(mapper, p.comp.langopts, p.strings.writer());
               ^
```

--- a/src/aro/Parser.zig
+++ b/src/aro/Parser.zig
@@ -6979,17 +6979,19 @@ fn unExpr(p: *Parser) Error!Result {
             {
                 if (tree.isBitfield(member_node)) try p.errTok(.addr_of_bitfield, tok);
             }
-            if (!tree.isLval(operand.node)) {
+            if (!tree.isLval(operand.node) and !operand.ty.is(.invalid)) {
                 try p.errTok(.addr_of_rvalue, tok);
             }
             if (operand.ty.qual.register) try p.errTok(.addr_of_register, tok);
 
-            const elem_ty = try p.arena.create(Type);
-            elem_ty.* = operand.ty;
-            operand.ty = Type{
-                .specifier = .pointer,
-                .data = .{ .sub_type = elem_ty },
-            };
+            if (!operand.ty.is(.invalid)) {
+                const elem_ty = try p.arena.create(Type);
+                elem_ty.* = operand.ty;
+                operand.ty = Type{
+                    .specifier = .pointer,
+                    .data = .{ .sub_type = elem_ty },
+                };
+            }
             try operand.saveValue(p);
             try operand.un(p, .addr_of_expr);
             return operand;

--- a/test/cases/binary expressions.c
+++ b/test/cases/binary expressions.c
@@ -72,6 +72,11 @@ int invalid_ptr_arithmetic(struct Foo *num) {
     return num - num;
 }
 
+void address_of_invalid_type(void) {
+    char x[64];
+    char *y = &(x&x);    
+}
+
 _Static_assert((1 ^ -1) == -2, "");
 _Static_assert((0 ^ 0) == 0, "");
 _Static_assert((-1 ^ 0) == -1, "");
@@ -104,3 +109,5 @@ _Static_assert((-2 ^ 2) == -4, "");
     "binary expressions.c:62:7: error: invalid operands to binary expression ('int' and 'float')" \
     "binary expressions.c:67:5: error: used type 'void' where arithmetic or pointer type is required" \
     "binary expressions.c:72:16: error: arithmetic on a pointer to an incomplete type 'struct Foo'" \
+    "binary expressions.c:77:18: error: invalid operands to binary expression ('char *' and 'char *')" \
+

